Modules/Server/Sources/PocketCastsServer/Public/ServerPodcastManager+Update.swift

```diff
@@ -104,8 +108,15 @@ extension ServerPodcastManager {
                 continue
             }
 
-            // for subscribed podcasts, only add older episodes, newer ones are handled by a refresh
+            // For subscribed podcasts, only add older episodes, newer ones are handled by a refresh
+            // so stuff like auto-download, auto add to up next works as expected
+            // However, if this is not a new episode we force update the podcast episodes
             if podcast.isSubscribed(), episodeDate.timeIntervalSinceNow >= podcast.latestEpisodeDate?.timeIntervalSinceNow ?? TimeInterval.greatestFiniteMagnitude {
+                if !missingEpisodesThatIsNotTheLastOne {
```

Comment:
This algorithm is terrible, it deserves a refactor. However, I'd like to get a review and thoughts about the idea before jumping into refactors/unit testing.

It just checks if the episodes that are missing are "in-between" or if they are the latest ones.